src/chain.cpp

```diff
@@ -108,3 +110,31 @@ void CBlockIndex::BuildSkip()
     if (pprev)
         pskip = pprev->GetAncestor(GetSkipHeight(nHeight));
 }
+
+void CBlockIndex::ClearSolution()
+{
+    AssertLockHeld(cs_main);
+    nSolution.clear();
+    nSolution.shrink_to_fit();
+}
+
+CBlockHeader CBlockIndex::GetBlockHeader() const
```

Comment:
Move these above `InvertLowestOne` so the order of functions in this file roughly matches the order in the header.